src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -97,6 +97,37 @@ public static SourceText From(string text, Encoding encoding = null, SourceHashA
             return new StringText(text, encoding, checksumAlgorithm: checksumAlgorithm);
         }
 
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
```

Comment:
There are already some global suppressions for other overloads. Consider doing the same for this one.